.github/workflows/update-docs.yml

```diff
@@ -0,0 +1,53 @@
+name: Update Template Documentation
+
+on:
+  schedule:
+    - cron: '0 16 * * 2' # Every Tuesday at 11 AM EST (16:00 UTC)
+  workflow_dispatch:
+
+jobs:
+  update-docs:
+    name: Update Template Documentation
+    runs-on: ubuntu-latest
+    if: github.repository == 'GoogleCloudPlatform/DataflowTemplates'
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: maven
+
+      - name: Install Templates Plugin
+        run: |
+          mvn clean install -pl plugins/templates-maven-plugin -am
+      - name: Generate Documentation
+        run: |
+          mvn clean prepare-package \
+            -DskipTests \
+            -PtemplatesSpec
+
+      - name: Check for changes
+        id: changes
+        run: |
+          git status --porcelain
+          if [[ -z $(git status --porcelain) ]]; then
+            echo "no_changes=true" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Create Pull Request
+        if: steps.changes.outputs.no_changes != 'true'
+        uses: peter-evans/create-pull-request@v7
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          commit-message: 'docs: update generated documentation'
+          title: 'chore(docs): update generated documentation'
+          body: |
+            This PR updates the generated documentation for the templates.
+            This is an automated PR created by the update-docs workflow.
```

Comment:
Checks won't auto-run when this PR is created, but we want them to run. We should add a line here like:

```
To review this PR, please (1) review the contents to make sure all doc updates are formatted correctly, (2) close and reopen the PR (3) approve/merge if tests pass successfully.
```

This is similar to what we do here - https://github.com/apache/beam/blob/31f9e5838f1eb488e5ab9a1d815e42a2eeb27b53/.github/workflows/update_python_dependencies.yml#L98C218-L98C403

Reply:
Yeah, I think that is probably fine. I was thinking about formatting, but it should get auto-picked up.

Eventually we might just want to commit to the `main` branch directly once we've seen this working well.